docs/source/api.rst

```diff
@@ -8,7 +8,18 @@ retworkx API
    the ``StableGraph`` type. The limitations and quirks with this library and
    type dictate how this operates. The biggest thing to be aware of when using
    the PyDAG class is that an integer node and edge index is used for accessing
-   elements on the DAG, not the data/weight of nodes and edges.
+   elements on the DAG, not the data/weight of nodes and edges. By default the
+   PyDAG realtime cycle checking is disabled for performance, however you can
+   opt-in to having the PyDAG class ensure that no cycles are added by setting
+   the ``check_cycle`` attribute to True. For example::
+
+       import retworkx
+       dag = retworkx.PyDAG()
+       dag.check_cycle = True
+
+   With check_cycle set to true any calls to :method:`PyDAG.add_edge` will
+   ensure that no cycles are added, ensuring that the PyDAG class truly
+   represents a directed acyclic graph.
```

Comment:
One possible source of confusion from this structure arises if a user toggles `check_cycle`. e.g. if a user sets `check_cycle=False`, introduces a cycle (on purpose or by mistake), sets `check_cycle=True`, edge additions will (currently) check that no additional cycles are introduced, but it is still possible the graph is cyclic.

If this is the desired behavior, we should document that the user is responsible for checking for existing cycles when setting `check_cycle=True` (and maybe find a more specific name).

Otherwise, the `check_cycle` setter could search for cycles when being set to `True`, or `check_cycle` could be a static property of an instance.

Reply:
Done in: https://github.com/mtreinish/retworkx/pull/28/commits/2b8e8ef80536f3eb3f750fbbbf57a1d70739d3ff